Update README.md and add TFLite validation process to auto_process.py (#6)
Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85: README.md

```
</br>

## 2. Prepare dataset
Download the STREETS dataset and annotations from [link](https://databank.illinois.edu/datafiles/ht2io/download), unzip, and move the vehicleannotaitons folder to ../dataset/ directory

```
Your code structure should like

├── datasets
│    ├── vehicleannotaitons
│    │    ├── images
│    │    └── annotations

> └── ModelZoo

Reviewer: ```suggestion
└── ModelZoo-YOLOFastest-for-ARM-U55-M85
```

Author: 수정하였습니다.

```
</br>

## 3. Training
If you want to start from scratch, create a '.pt' file via 'train.py'.
```bash

> python train.py --data ./data/streets.yaml --epochs 300 --weights '' --cfg ./models/yolo-fastest.yaml  --batch-size 512

Reviewer: ```suggestion
python train.py --data ./data/STREETS.yaml --epochs 300 --weights '' --cfg ./models/yolo-fastest.yaml  --batch-size 128
```

Author: 여기서 train.py 실행할 때 batch size 512으로 진행했는데요 128으로 수정해될까요

Author: 어제 저녁 마지막 학습 했을때 는 8개 gpu를 512으로 학습하였습니다. 64으로 수정하도록 하겠습니다.

Author: 어제 저녁 마지막 학습 했을때 는 8개 gpu, batch size 512으로 학습하였습니다.
64으로 수정하도록 하겠습니다.

```
</br>

## 4. Compress model, convert to tflite, and benchmark with PyNetsPresso

`auto_process.py` provides integrated process which contains torch.fx converting, model compression, fx model retraining, onnx exporting, tflite converting, device benchmark, and mAP validation. You can execute `auto_process.py` with minimal training hyper-parameters and NetsPresso account information.

You can choose Renesas-RA8D1 (Arm Cortex-M85) or Ensemble-E7-DevKit-Gen2 (Arm Cortex-M55 + Ethos-U55) device, and boost inference speed by giving Helium option.
``` bash

> python auto_process.py --data ./data/streets.yaml --name yolo_fastest --weight_path ./models/yolo_fastest_streets.pt --epochs 300 --batch-size 512 --np_email '' --np_password '' --target_device Renesas-RA8D1 --helium

Reviewer: ```suggestion
python auto_process.py --data ./data/STREETS.yaml --name yolo_fastest --weight_path ./models/yolo_fastest_streets_256.pt --epochs 300 --batch-size 128 --np_email '' --np_password '' --target_device Renesas-RA8D1 --helium
```

batch size는 4 gpu 기준으로 512 작성한거면 single gpu 기준으로 128로 바꿀까요?

Author: 어제 저녁 마지막 학습 했을때 는 8개 gpu, batch size 512으로 학습하였습니다.
64으로 수정하도록 하겠습니다.

| [YOLO-Fastest](https://github.com/Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85/tree/master/models/yolo_fastest_uadetrac_256.pt)           |FP32   | 256                   | 41.6                | 75.5              | **-**                       | **-**                       | **-**                        | **0.3**            |
[YOLO-Fastest TFLite](https://github.com/Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85/tree/master/models/yolo_fastest_uadetrac_full_int8_256.tflite)     | Full INT8         | 256              | 39.7           | 72.8              | **513**                       | **234**                       | **6.7**                        | **0.3**            |
<details>
  <summary>Table Notes</summary>

- The checkpoint is trained to 300 epochs with default settings. The model uses [hyp.scratch-low.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyps/hyp.scratch-low.yaml) hyps.

> - **mAP<sup>val</sup>** values are for single-model single-scale on [STREETS](https://databank.illinois.edu/datasets/IDB-3671567) dataset.<br>Reproduce by `python val.py --weights './models/yolo_fastest_streets_256.pt' --data ./data/streets.yaml --img 256` for pytorch ckpt file, and

Reviewer: ```suggestion
- **mAP<sup>val</sup>** values are for single-model single-scale on [STREETS](https://databank.illinois.edu/datasets/IDB-3671567) dataset.<br>Reproduce by `python val.py --weights './models/yolo_fastest_streets_256.pt' --data ./data/STREETS.yaml --img 256` for pytorch ckpt file, and
```

Author: streets -> STREETS으로 수정하였습니다.

>  `python val.py --weights './models/yolo_fastest_streets_full_int8_256.tflite' --data ./data/UA-DETRAC.yaml --img 256--anchors-for-tflite-path /ssd1/tairen.piao/nota_github/ModelZoo-YOLOFastest-for-ARM-U55-M85/models/yolo_fastest_streets_256_anchors.json` for full int8 tflite file.

Reviewer: ```suggestion
 `python val.py --weights './models/yolo_fastest_streets_full_int8_256.tflite' --data ./data/UA-DETRAC.yaml --img 256 --anchors-for-tflite-path './models/yolo_fastest_streets_256_anchors.json` for full int8 tflite file.
```

int8 tflite file의 이름의 경우 auto_process.py를 실행하면 어떻게 생성되나요? 그 파일 이름과 맞출 필요가 있습니다.
anchors.json file의 이름도 마찬가지로 자동 저장되는 파일 이름과 맞는지 확인부탁드립니다.

Author: 여기서 질문 있습니다. benchmark 테이블에 INT8 TFLite는 압축 안 한 모델의 성능이라서 이렇게 작성하였습니다. 어떻게 진행하면 좋을까요?

Reviewer: @tairenpiao 데일리미팅에서 말씀드린 것처럼 압축 전/후의 mAP, latency를 공유해주시면 어떻게 진행할지 답변드리겠습니다.

Author: 말씀하신 대로 압축된 모델의 mAP, latency 정보를 업데이트 하였습니다. 확인해 주시면 감사하겠습니다.
compressed tflite 파일은 아직 업로드 하지 않았습니다. 결정되는 대로 수정하도록 하겠습니다.

Reviewer: 정보 전달 측면에서 수정하신 방향으로 진행하시죠.
compressed tflite 파일은 업로드 하지 않아도 될 것 같습니다.

Benchmark table에 model 오른쪽에 column하나 추가해서 format으로 기입하고 그 아래에 pytorch, tflite라고 기입하면 어떨까요? 그러면 model name에 있는 tflite를 뺄 수 있을 것 같습니다.

Author: 말씀하신 대로 수정하였습니다.

## 2. Training
If you want to start from scratch, create a '.pt' file via 'train.py'.
```bash

> python train.py --data ./data/UA-DETRAC.yaml --epochs 300 --weights '' --cfg ./models/yolo-fastest.yaml  --batch-size 32

Reviewer: `--epoch` 300으로 진행하셨나요? 아니면 더 적게 하셨나요?
`models/yolo-fastest.pt`가 COCO pretrained weight인데요. `--weights`를 위해 이것 사용 안하고 scratch로 학습하셨나요?

Author: streets dataset을 학습할 때 coco pre-trained weight를 사용하고 300 epochs으로 진행하였습니다.

```
</br>

> ## 3. Compress model and export to tflite with PyNetsPresso

Reviewer: ```suggestion
## 3. Compress model, convert to tflite, and benchmark with PyNetsPresso
```

`auto_process.py` provides integrated process which contains torch.fx converting, model compression, fx model retraining, onnx exporting, tflite converting, device benchmark, and mAP validation. You can execute `auto_process.py` with minimal training hyper-parameters and NetsPresso account information.

You can choose Renesas-RA8D1 (Arm Cortex-M85) or Ensemble-E7-DevKit-Gen2 (Arm Cortex-M55 + Ethos-U55) device, and boost inference speed by giving Helium option.
``` bash

> python auto_process.py --data ./data/UA-DETRAC.yaml --name yolo_fastest --weight_path ./models/yolo_fastest_uadetrac_256.pt --epochs 300 --batch-size 32 --np_email '' --np_password '' --target_device Renesas-RA8D1 --helium

Reviewer: 여기서도 `--epochs` 300이 아닌 다른 값으로 진행하셨다면 변경 부탁드립니다.

```
</br>

## Benchmark

|Model                                                                                           | Precision| Size<br><sup>(pixels) | mAP<sup>val<br>50-95 | mAP<sup>val<br>50 | Speed<br><sup>Cortex-M85<br>(ms) | Speed<br><sup>Cortex-M85 with Helium<br>(ms) | Speed<br><sup>Ethos-U55<br>(ms) | Params<br><sup>(M) |
| ----------------------------------------------------------------------------------------------- | -----|--------------------- | -------------------- | ----------------- | ---------------------------- | ----------------------------- | ------------------------------ | ------------------ |

> | [YOLO-Fastest ckpt](https://github.com/Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85/tree/master/models/yolo_fastest_uadetrac_256.pt)           |FP32   | 256                   | 23.5                 | 42.1              | **-**                       | **-**                       | **-**                        | **0.3**            |

Reviewer: ```suggestion
| [YOLO-Fastest](https://github.com/Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85/tree/master/models/yolo_fastest_uadetrac_256.pt)           |FP32   | 256                   | 23.5                 | 42.1              | **-**                       | **-**                       | **-**                        | **0.3**            |
```

Author: 수정하였습니다.

[YOLO-Fastest](https://github.com/Nota-NetsPresso/ModelZoo-YOLOFastest-for-ARM-U55-M85/tree/master/models/yolo_fastest_streets_full_int8_256.tflite)     | TFLite|Full INT8         | 256              | 39.7           | 72.8              | **594**                       | **269**                       | **6.8**                        | **0.3**            |
Compressed YOLO-Fastest    |TFLite | Full INT8         | 256              | 37.3           | 71.5              | **513**                       | **234**                       | **6.0**                        | **0.2**            |
<details>
  <summary>Table Notes</summary>

- The checkpoint is trained to 300 epochs with default settings. The model uses [hyp.scratch-low.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyps/hyp.scratch-low.yaml) hyps.
- **mAP<sup>val</sup>** values are for single-model single-scale on the [STREETS](https://databank.illinois.edu/datasets/IDB-3671567) dataset.<br>Reproduce by `python val.py --weights './models/yolo_fastest_streets_256.pt' --data ./data/STREETS.yaml --img 256` for pytorch ckpt file, and

>  `python val.py --weights './models/yolo_fastest_streets_full_int8_256.tflite' --data ./data/STREETS.yaml --img 256 --anchors-for-tflite-path /ssd1/tairen.piao/nota_github/ModelZoo-YOLOFastest-for-ARM-U55-M85/models/yolo_fastest_streets_256_anchors.json` for full int8 tflite file.

Reviewer: /ssd1/tairen.piao/nota_github/ModelZoo-YOLOFastest-for-ARM-U55-M85/models/yolo_fastest_streets_256_anchors.json

-> ./models/yolo_fastest_streets_256_anchors.json

Author: 좋은 코멘트 감사합니다.
수정하였습니다.
